lib/improver/weights.py

```diff
@@ -451,3 +451,110 @@ def __repr__(self):
+                .
+        """
+
+        def calculate_weight(point, slope):
+            """
+            A helper function to calculate the weights for each point using a
+            piecewise function to build up the triangular function.
+            Args:
+                point : float, the point in the coordinate from the cube for
+                        which we want to calculate a weight for.
+                slope : float, the gradient of the triangle, calculated from
+                        1/(width of triangle).
+            """
+            if point < self.midpoint:
+                weight = 1-(self.midpoint-point)*slope
+            if point == self.midpoint:
+                weight = 1
+            if point > self.midpoint:
+                weight = 1-(point-self.midpoint)*slope
+            return weight
+
+        slope = 1.0/self.width
+        weights = np.zeros(coord_vals.shape)
+        # Find the indices of the points where there will be non-zero weights.
+        condition = ((coord_vals >= (self.midpoint-self.width)) &
+                     (coord_vals <= (self.midpoint+self.width)))
+        points_with_weights = np.where(condition)[0]
+        # Calculate for weights for points where we want a non-zero weight.
+        for index in points_with_weights:
+            weights[index] = calculate_weight(coord_vals[index], slope)
+        # Normalise the weights.
+        weights = WeightsUtilities.normalise_weights(weights)
+
+        return weights
+
+    def process(self, cube, coord_name):
+        """Calculate triangular weights for a given cube and coord.
+
+            Args:
+                cube : iris.cube.Cube
+                       Cube to blend across the coord.
+                coord_name : string
+                       Name of coordinate in the cube to be blended.
+
+            Returns:
+                weights : array of weights, sum of all weights = 1.0
+
+            Raises:
+                ValueError : input is not a cube
+        """
+        if not isinstance(cube, iris.cube.Cube):
+            msg = ('The first argument must be an instance of '
+                   'iris.cube.Cube but is'
+                   ' {0:s}'.format(type(cube)))
+            raise ValueError(msg)
+
+        cube_coord = cube.coord(coord_name)
+        coord_vals = cube_coord.points
+        coord_units = cube_coord.units
+
+        # Rescale width and midpoint if in differnt units to the coordinate
+        if coord_units != self.parameters_units:
+            self.width = self.parameters_units.convert(self.width, coord_units)
+            self.midpoint = self.parameters_units.convert(self.midpoint,
+                                                          coord_units)
+
+        weights = self.triangular_weights(coord_vals)
+
+        return weights
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string."""
+        string = ("<ChooseDefaultTriangularWeights width={0:4.1f},"
```

Comment:
Remove leading zeroes from format statements to make it print the correct information.